lib/stripe/api_resource.rb

```diff
@@ -48,6 +48,30 @@ def self.save_nested_resource(name)
       end
     end
 
+    # Adds a custom method to a resource class. This is used to add support for
+    # non-CRUDL API requests, e.g. capturing charges. custom_method takes the
+    # following parameters:
+    # - name: the name of the custom method to create (as a symbol)
+    # - http_verb: the HTTP verb for the API request (:get, :post, or :delete)
+    # - http_path: the path to append to the resource's URL. If not provided,
+    #              the name is used as the path
+    #
+    # For example, this call:
+    #     custom_method :capture, http_verb: post
+    # adds a `capture` class method to the resource class that, when called,
+    # will send a POST request to `/v1/<object_name>/capture`.
+    def self.custom_method(name, http_verb:, http_path: nil)
+      unless %i[get post delete].include?(http_verb)
+        raise ArgumentError, "Invalid http_verb value: #{http_verb.inspect}. Should be one of :get, :post or :delete."
```

Comment:
The error message is wrong right? It should **not** be one of those I think.

Reply:
I think you're confusing CRUDL methods with HTTP verbs. All requests to Stripe's API (CRUDL or custom) use either GET, POST or DELETE as the HTTP verb, so it has to be one of those three.